packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/version.dart';
+
+// VS Code layout:
+
+// macOS:
+//   /Applications/Visual Studio Code.app/Contents/
+//   /Applications/Visual Studio Code - Insiders.app/Contents/
+//   $HOME/Applications/Visual Studio Code.app/Contents/
+//   $HOME/Applications/Visual Studio Code - Insiders.app/Contents/
+// macOS Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+// Windows:
+//   $programfiles(x86)\Microsoft VS Code
+//   $programfiles(x86)\Microsoft VS Code Insiders
+// TODO: Confirm these are correct for 64bit
+//   $programfiles\Microsoft VS Code
+//   $programfiles\Microsoft VS Code Insiders
+// Windows Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+// Linux:
+//   /usr/share/code/bin/code
+//   /usr/share/code-insiders/bin/code-insiders
+// Linux Extensions:
+//   $HOME/.vscode/extensions
+//   $HOME/.vscode-insiders/extensions
+
+const String extensionIdentifier = 'Dart-Code.dart-code';
```

Comment:
kExtensionIdentifier per flutter style guide https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#begin-global-constant-names-with-prefix-k

Reply:
I made the constants non-global instead (96330f6), seemed like they didn't need to be (if I understood the style guide correctly, they don't need prefixing this way?).